tests/test_file.py

```diff
@@ -624,3 +633,62 @@ def test_file_zarr():
     file.save()
     file.delete(storage=False)
     UPath("test.zarr").unlink()
+
+
+def test_zarr_folder_upload():
+    previous_storage = ln.setup.settings.storage.root_as_str
+    ln.settings.storage = "s3://lamindb-test"
+
+    def callback(*args, **kwargs):
+        pass
+
+    zarr_path = Path("./test_adata.zrad")
+    write_adata_zarr(adata, zarr_path, callback)
+
+    file = ln.File(zarr_path, key="test_adata.zrad")
+    file.save()
+
+    assert isinstance(file.path, CloudPath) and file.path.exists()
+
+    file.delete(storage=True)
+    delete_storage(zarr_path)
+    ln.settings.storage = previous_storage
+
+
+def test_df_suffix():
+    file = ln.File(df, key="test_.parquet")
+    assert file.suffix == ".parquet"
+
+    with pytest.raises(ValueError) as error:
+        file = ln.File(df, key="test_.def")
+    assert (
+        error.exconly().partition(",")[0]
+        == "ValueError: The suffix '.def' of the provided key is incorrect"
+    )
+
+
+def test_adata_suffix():
```

Comment:
Can you please use `pytest.mark.parametrize`, here? Otherwise it's hard to read and to assess completeness.

Reply:
Different combinations can raise different errors or require different asserts, that is why not parametrized. Parametrization here becomes a mess actually in my opinion.

Reply:
Ok, then! No parametrization!